Dynavity/Dynavity/persistence/storage/CloudStorageManager.swift

```diff
@@ -135,6 +136,40 @@ struct CloudStorageManager: OnlineStorageManager {
         // return imported canvas
         return canvas
     }
+
+    func addChangeListeners(model: OnlineCanvas) {
```

Comment:
Can this take in an `OnlineCanvasDTO` and perform the changes on the DTO instead?
So that the actual persistence implementation does not know about the model

Reply:
I don't think this will work since the purpose of change listeners is to directly update the `OnlineCanvas` state that is used by `CanvasViewModel`. The `OnlineCanvasDTO` will not be able to do this as it has no association to the model once `toModel` finishes.

An alternative (instead of calling `OnlineCanvasDTO` -> `CanvasRepository` -> `CloudStorageManager`) is to try and set up the listeners when `CloudStorageManager` initialises `OnlineCanvasDTO`. However, in this approach all the observer setup needs to be deferred until the model object is created, using copious amounts of escaping closures.